### PR TITLE
Fetch Data from Endpoint as BuildSnapshotContainer Mounts

### LINKED ISSUE #17: Fetch from REST API on front-end
Interact with REST API from the front-end to fetch BuildSnapshot data.

Requirement Reference: 7

### PR DESCRIPTION
**Pull request by samuelsogbesan**
This PR integrates the data fetch from the front-end. 

When the BuildSnapshotContainer (#47) component is [mounted (the useEffect Hook)](https://reactjs.org/docs/hooks-effect.html), a GET request is made to an endpoint to retrieve our build data. Map Recursion is performed on this array of JSON data to create a corresponding BuildSnapshot (#76) Component.

_Note, useEffect takes a second argument, which are the dependent variables.  [We pass in an empty array](https://www.robinwieruch.de/react-hooks-fetch-data) to prevent the hook from re-firing as the component is reloaded / updated, which would cause an infinite callback (since the updates would propagate and cause more updates)_.

For caching and performance optimisation, we [memoize](https://reactjs.org/docs/react-api.html#reactmemo) our container so that it will not re-render if our input information doesn't change.

Closes #17

### CODE REVIEW

File: src/main/webapp/components/BuildSnapshotContainer.js [hunk 1]
```diff
@@ -8,11 +8,25 @@ import {Wrapper} from "./Wrapper";
...
-      {data.map(snapshotData => <BuildSnapshot data={snapshotData}/>)}
-    </Wrapper>
-  );
-}
+export const BuildSnapshotContainer = React.memo((props) => {
+    const SOURCE = '/data';
+    const [data, setData] = React.useState([]);
+
+    /**
+    * Fetch data when component is mounted.
```
Comment (krasimir-google) [inline src/main/webapp/components/BuildSnapshotContainer.js:16]: nit: please add a space before the `*` here and below to align the column of stars in the comment with the first start from the `/**` line.

**Review by krasimir-google — CHANGES_REQUESTED** [review]

File: src/main/webapp/components/BuildSnapshotContainer.js (hunk 1 above)
Comment (samuelsogbesan) [reply src/main/webapp/components/BuildSnapshotContainer.js]: Updated alignment here 48edcfe

File: src/main/webapp/components/BuildSnapshotContainer.js [hunk 2]
```diff
@@ -8,11 +8,25 @@ import {Wrapper} from "./Wrapper";
...
-export const BuildSnapshotContainer = (props) => {
-  const data = props.data;
-  return (
-    <Wrapper>
-      {data.map(snapshotData => <BuildSnapshot data={snapshotData}/>)}
-    </Wrapper>
-  );
-}
+export const BuildSnapshotContainer = React.memo((props) => {
+    const SOURCE = '/data';
```
[the remark this answers is not in the document]
Comment (samuelsogbesan) [reply src/main/webapp/components/BuildSnapshotContainer.js:13]: Update here e105d18. Is this valid formatting?

[the remark this answers is not in the document]
Comment (krasimir-google) [reply src/main/webapp/components/BuildSnapshotContainer.js:13]: Yes, I missed the fact that there were 2 levels of indirection.

**Review by krasimir-google — APPROVED** [review]
Thank you! LGTM

**Pull request merged**